platform/projectModel-impl/src/com/intellij/openapi/module/impl/UnloadedModuleDescriptionImpl.kt

```diff
@@ -45,16 +46,17 @@ class UnloadedModuleDescriptionImpl(val modulePath: ModulePath,
   companion object {
     @JvmStatic
     fun createFromPaths(paths: Collection<ModulePath>, parentDisposable: Disposable): List<UnloadedModuleDescriptionImpl> {
+      val interner = HashSetInterner<String>()
       val pathVariables = JpsGlobalLoader.computeAllPathVariables(PathManager.getOptionsPath())
       val modules = JpsProjectLoader.loadModules(paths.map { Paths.get(it.path) }, null, pathVariables, JpsPathMapper.IDENTITY)
-      val pathsByName = paths.associateBy { it.moduleName }
-      return modules.map { create(pathsByName[it.name]!!, it, parentDisposable) }
+      val pathsByName = paths.associateBy { interner.intern(it.moduleName) }
```

Comment:
Why do you need to intern string here? The returned value is used inside `pathsByName` map which doesn't escape from this method.